src/posix/platform/system.cpp

```diff
@@ -100,8 +101,15 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
 #endif
 
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
-    // Reuse the backbone interface name.
-    platformInfraIfInit(instance, aPlatformConfig->mBackboneInterfaceName);
+    {
+        // Reuse the backbone interface name.
+        uint32_t infraIfIndex = platformInfraIfInit(instance, aPlatformConfig->mBackboneInterfaceName);
+
+        SuccessOrDie(otBorderRoutingInit(instance, infraIfIndex));
+        otPlatInfraIfStateChanged(instance, infraIfIndex, platformInfraIfIsRunning(),
+                                  platformInfraIfGetLinkLocalAddress());
+        IgnoreError(otBorderRoutingSetEnabled(instance, /* aEnabled */ true));
```

Comment:
why ignore error?

Reply:
We are sure this will success. `OT_ASSERT` will be verbose.

Reply:
replace with `SuccessOrDie`.